### PR TITLE
Tsify some tests

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
I've converted a handful of test cases. Draft at the moment.

### CODE REVIEW

File: test/base-formatter-tests.ts [hunk 1]
```diff
@@ -24,11 +24,13 @@
 
 import {BaseFormatter} from '../lib/formatters/base';
 
+class Formatter extends BaseFormatter {}
```
Comment (jeremy-rifkin) [inline test/base-formatter-tests.ts:27]: TS refuses to instantiate an abstract class so just doing this

File: test/base-formatter-tests.ts [hunk 2]
```diff
@@ -24,11 +24,13 @@
...
+
 describe('Basic formatter functionality', () => {
     it('should be one-true-style if the styles are empty', () => {
-        const fmt = new BaseFormatter({
+        const fmt = new Formatter({
             name: 'foo-format',
-            exe: null,
+            exe: '/dev/null',
```
Comment (jeremy-rifkin) [inline test/base-formatter-tests.ts:33]: Saw `exe: '/dev/null'` in another file and figured it was a good solution

File: test/utils.ts [hunk 3]
```diff
@@ -39,6 +41,16 @@ export function makeCompilationEnvironment(options: Record<string, any>): Compil
     return new CompilationEnvironment(compilerProps, compilationQueue, options.doCache);
 }
 
+export function makeFakeCompilerInfo(props: Partial<CompilerInfo>): CompilerInfo {
```
Comment (jeremy-rifkin) [inline test/utils.ts:44]: This isn't super elegant but it gets the job done. There's another `makeFakeCompilerInfo` in other parts of the code that should be merged with this at some point.

File: test/exec-tests.ts [hunk 4]
```diff
@@ -188,11 +189,14 @@ describe('Execution tests', () => {
             expect(filenameTransform).to.be.undefined;
         });
         it('should pass through options', () => {
-            const options = exec.getNsJailOptions('sandbox', '/path/to/compiler', [], {some: 1, thing: 2}).options;
-            options.should.deep.equals({some: 1, thing: 2});
+            const options = exec.getNsJailOptions('sandbox', '/path/to/compiler', [], {
```
Comment (jeremy-rifkin) [inline test/exec-tests.ts:192]: Now with typing we need to pass options that actually exist

File: test/utils.ts (hunk 3 above)
Comment (mattgodbolt) [inline test/utils.ts:44]: well that's pretty fake hahah :) nice

**Review by mattgodbolt — APPROVED** [review]
Thanks for spearheading this!

**Comment by jeremy-rifkin** [comment]
Thanks for the review!

**Pull request merged**